### PR TITLE
fix: retry curl command on transient errors

### PR DESCRIPTION
**Pull request by sreis**
Follow up to [this review request](https://github.com/opstrace/opstrace/pull/94#discussion_r535627347).

### CODE REVIEW

File: ci/utils.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#
+# If this option is used several times, the last one will be used."
+#
+curl() {
+    # circumvent normal bash function lookup
+    command curl \
+    --connect-timeout 10 \
+    --retry 3 \
+    --retry-delay 5 \
+    "${@}"
```
Comment (jgehrcke) [inline ci/utils.sh:34]: Yes!!! :rocket:

Comment (jgehrcke) [reply ci/utils.sh:34]: @sreis do we also want to add verbosity? For example, when curl retries now upon a 5xx HTTP response -- maybe we want to log that. On the other hand, we can try to remember adding `-v` in those places where we use this curl shim.

**Review by jgehrcke — APPROVED** [review]
This is great; and also necessary towards getting CI stability into a nice regime.

Next up: docker pull, docker push... haha.

File: ci/utils.sh (hunk 1 above)
Comment (sreis) [reply ci/utils.sh:34]: It prints a `warning` by default. Example message for a timeout:
```Warning: Transient problem: timeout Will retry in 1 seconds. 3 retries left.```

Comment (jgehrcke) [reply ci/utils.sh:34]: Perfect.

**Pull request merged**